server/welcomebot.go

```diff
@@ -45,6 +45,37 @@ func (p *Plugin) constructMessageTemplate(userID, teamID string) *MessageTemplat
 	return data
 }
 
+func (p *Plugin) constructChannelMessageTemplate(userID, channelID string) *ChannelMessageTemplate {
+	data := &ChannelMessageTemplate{}
+	var err *model.AppError
+
+	if len(userID) > 0 {
+		if data.User, err = p.API.GetUser(userID); err != nil {
+			p.API.LogError("failed to query user", "UserID", userID)
+			return nil
+		}
+	}
+
+	if data.User != nil {
+		if data.DirectMessage, err = p.API.GetDirectChannel(userID, p.botUserID); err != nil {
+			p.API.LogError("failed to query direct message channel", "UserID", userID)
+			return nil
+		}
+	}
+
+	if len(channelID) > 0 {
+		data.Channel, err = p.API.GetChannel(channelID)
+		if err != nil {
+			p.API.LogError("failed to query channel", "ChannelID", channelID)
+			return nil
+		}
+	}
+
+	data.UserDisplayName = data.User.GetDisplayName(model.ShowNicknameFullName)
+
+	return data
```

Comment:
Is it the case that `channelID` will _always_ be provided and non-empty? Is it the case that this function must return a fully populated `ChannelMessageTemplate`?

If that's the case, we should return an error if necessary passed in data is not available. Not sure about the situation here though

Reply:
No, it is not necessary to pass the channel ID here. Also, the function doesn't need to return a fully populated `ChannelMessageTemplate`.